### PR TITLE
Rename metric and indicator to metrics.

### PR DESCRIPTION
**Pull request by peng-yongsheng**
If we say our project is a observability system. It contains 3 core concept: Logging, Metrics, Tracing. So, we should unify the nouns of metrics.

### CODE REVIEW

**Comment by wu-sheng** [comment]
@kezhenxu94 This pr includes concept updates, a little related to the document.

File: oap-server/generate-tool/src/main/java/org/apache/skywalking/oal/tool/parser/Metrics.java [hunk 1]
```diff
@@ -22,31 +22,30 @@
 import com.google.common.reflect.ClassPath;
 import java.io.IOException;
 import java.util.*;
-import org.apache.skywalking.oap.server.core.analysis.indicator.Indicator;
-import org.apache.skywalking.oap.server.core.analysis.indicator.annotation.IndicatorFunction;
+import org.apache.skywalking.oap.server.core.analysis.metrics.annotation.MetricsFunction;
 
-public class Indicators {
-    private static Map<String, Class<? extends Indicator>> REGISTER = new HashMap<>();
+public class Metrics {
```
Comment (wu-sheng) [inline oap-server/generate-tool/src/main/java/org/apache/skywalking/oal/tool/parser/Metrics.java]: Could we change this to MetricsHolder? I don't want to use full name below.

**Comment by kezhenxu94** [comment]
> @kezhenxu94 This pr includes concept updates, a little related to the document.

SkyAPM/document-cn-translation-of-skywalking/issues/23

**Comment by wu-sheng** [comment]
@flycash Please notice, `NamespacedPodListWatchTest` test has been deleted in this pull request, because it triggers k8s java client bug, https://github.com/kubernetes-client/java/issues/460

We could add this bug after our k8s client upgrade later.

**Pull request merged**